### PR TITLE
Fix nx.from_pandas_edgelist so edge keys are not added as edge attributes and edge keys

### LINKED ISSUE #7444: keys added as edge attributes when using nx.from_pandas_edgelist
Hi there,
regarding the nx.from_pandas_edgelist function, when creating multigraphs:

### Current Behavior

Following the example in the documentation, but calling "edge_attr = True" instead of "edge_attr=["weight", "color"]", the edge keys get added to the graph not only as keys, but also as edge attributes.
i.e G.edges(data=True) outputs:
`MultiEdgeDataView([(0, 2, {'my_edge_key': 'A', 'weight': 3, 'color': 'red'}), (0, 2, {'my_edge_key': 'D', 'weight': 6, 'color': 'blue'}), (2, 1, {'my_edge_key': 'B', 'weight': 4, 'color': 'blue'}), (2, 3, {'my_edge_key': 'C', 'weight': 5, 'color': 'blue'})])`
I do not think this is intended behavior, as edge keys are not edge attributes. 
In other words If we start with a multigraph and save it to a pandas edgelist using `nx.to_pandas_edgelist()`, the graph we load from that same edgelist in this way has an extra attribute (my_edge_key).

### Expected Behavior
keys should only be added as keys, as is what happens when you manually specify which columns are edge_attr.

### Steps to Reproduce

`edges = pd.DataFrame(
    {
        "source": [0, 1, 2, 0],
        "target": [2, 2, 3, 2],
        "my_edge_key": ["A", "B", "C", "D"],
        "weight": [3, 4, 5, 6],
        "color": ["red", "blue", "blue", "blue"],
    }
)`

`
G = nx.from_pandas_edgelist(
    edges,
    edge_key="my_edge_key",
    edge_attr=["weight", "color"],
    create_using=nx.MultiGraph(),
)
`

### Environment

NetworkX version: 3.2.1

### PR DESCRIPTION
**Pull request by inbalh1**
Fixes #7444

### CODE REVIEW

File: networkx/convert_matrix.py [hunk 1]
```diff
@@ -348,7 +348,7 @@ def from_pandas_edgelist(
     edge_attr : str or int, iterable, True, or None
         A valid column name (str or int) or iterable of column names that are
         used to retrieve items and add them to the graph as edge attributes.
-        If `True`, all of the remaining columns will be added.
+        If `True`, all of the remaining columns will be added (not including edge_key).
```
Comment (rossbar) [inline networkx/convert_matrix.py:351]:
```suggestion
        If `True`, all of the remaining columns will be added (not including `edge_key`).
```

**Review by rossbar — APPROVED** [review]
LGTM, thanks @inbalh1 ! Just one docstring formatting nit but the proposed solution looks good - clear, concise, and tested!

**Comment by inbalh1** [comment]
Thanks @rossbar !

File: networkx/convert_matrix.py [hunk 2]
```diff
@@ -348,7 +348,7 @@ def from_pandas_edgelist(
     edge_attr : str or int, iterable, True, or None
         A valid column name (str or int) or iterable of column names that are
         used to retrieve items and add them to the graph as edge attributes.
-        If `True`, all of the remaining columns will be added.
+        If `True`, all of the remaining columns will be added (not including `edge_key`).
```
Comment (dschult) [inline networkx/convert_matrix.py:351]: Is this better wording?  (it is <= 88 chars -- which is why I first started looking, but the end result is an attempt to allow the user to know what's going on without having to read all the other kwargs.
```suggestion
        If `True`, all columns will be added except `source`, `target` and `edge_key`.
```

**Review by dschult — COMMENTED** [review]
I also have a tweak of the doc_string. And I agree that this is a good solution.
Thanks!!

File: networkx/convert_matrix.py (hunk 2 above)
Comment (inbalh1) [reply networkx/convert_matrix.py]: Looks better to me, thanks @dschult

**Review by dschult — APPROVED** [review]
LGTM!  Thanks!

**Pull request merged**